bagit/validate_activity.go

```diff
@@ -0,0 +1,56 @@
+package bagit
+
+import (
+	"context"
+
+	"go.artefactual.dev/tools/temporal"
+)
+
+const ValidateActivityName = "validate-bag-activity"
+
+type BagValidator interface {
+	Validate(path string) error
+}
+
+type ValidateActivity struct {
+	cfg       *Config
+	validator BagValidator
+}
+
+func NewValidateActivity(cfg Config, validator BagValidator) *ValidateActivity {
+	cfg.setDefaults()
+
+	return &ValidateActivity{cfg: &cfg, validator: validator}
+}
+
+type ValidateActivityParams struct {
+	// Path is the full path of the Bag to be validated.
+	Path string
+}
+
+type ValidateActivityResult struct {
+	// Valid is true if the Bag is valid.
+	Valid bool
+
+	// Error is a message indicating why validation failed, and will always be
+	// empty when IsValid is true.
+	Error string
+}
+
+// Execute validates the BagIt Bag located at Path.
+func (a *ValidateActivity) Execute(
+	ctx context.Context,
+	params *ValidateActivityParams,
+) (*ValidateActivityResult, error) {
+	logger := temporal.GetLogger(ctx)
+	logger.V(1).Info("Executing BagIt Validate Activity", "Path", params.Path)
+
+	if err := a.validator.Validate(params.Path); err != nil {
+		return &ValidateActivityResult{
+			Valid: false,
+			Error: err.Error(),
+		}, nil
+	}
+
+	return &ValidateActivityResult{Valid: true}, nil
```

Comment:
I see https://github.com/artefactual-labs/bagit-gython doesn't make a distinction between validation and other type of errors, so I guess we could do the same and just return an empty result and the error from `Validate`. I'd prefer to look for the error directly when calling the activity, which will still be needed, than looking for that error inside the result.

Reply:
Yes, when I was looking at Go validation libraries it seemed like it was pretty common to return a validation error as a Go `error`, so I think it's consistent for https://github.com/artefactual-labs/bagit-gython to do the same.  I do think it's problematic that bagit-gython doesn't return a specialized error type to allow differentiating validation errors from application errors though — I think I'll file an issue ticket about that.

I'm inclined to keep the validation "failures" in the `ValidateActivityResult` for a few reasons though:

1. A validation error is NOT an application error, it's an expected outcome of trying to validate an invalid bag.  If validation fails it shouldn't trigger activity retries, and I don't think it should be logged as an activity error.   I know we can set the activity retry policy to not retry on a validation error, but I think it's better not to have to rely on a caller remembering to do that.
2. While we can use Go error type checking to differentiate validation errors from application errors, I'd rather not. Go error type checking is terrible in my opinion — it's one of my least favourite parts of the language. :(  I really want `errors.Is()` to just *work* with dynamic errors, but it just doesn't.  And I think doing type checking with `errors.As()` is ugly and counter-intuitive, e.g. 
  ```go
  var e bagit.ErrValidation
  if (errors.As(e, err)) { return "blah!" }
  ```